### PR TITLE
TestEngine: Get instruction coverage and move to net standard

### PR DESCRIPTION
**Pull request by shargon**
Part of https://github.com/neo-project/neo-devpack-dotnet/issues/889

This first version calculates coverage at the instruction level, it does not require source code to know how much a method has been covered, although you still do not know what parts of the method have been covered

### CODE REVIEW

**Comment by shargon** [comment]
@cschuchardt88 @Jim8y

**Comment by cschuchardt88** [comment]
@shargon can you fix format?

File: src/Neo.SmartContract.Testing/Coverage/CoverageBase.cs [hunk 1]
```diff
@@ -48,5 +48,15 @@ public IEnumerable<CoverageData> GetCoverageFrom(int offset, int length)
...
         }
+
+        // Allow to sum coverages
+
+        public static CoverageBase? operator +(CoverageBase? a, CoverageBase? b)
+        {
+            if (a is null) return b;
+            if (b is null) return a;
+
+            return new CoveredCollection(a, b);
```
Comment (Jim8y) [inline src/Neo.SmartContract.Testing/Coverage/CoverageBase.cs:67]: weird but also elegent.

File: tests/Neo.SmartContract.Testing.UnitTests/Coverage/CoverageDataTests.cs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+|                                registerCandidate,1 |    0.00% |
+|                                   setGasPerBlock,1 |    0.00% |
+|                                 setRegisterPrice,1 |    0.00% |
+|                                           symbol,0 |    0.00% |
+|                                      totalSupply,0 |  100.00% |
+|                                         transfer,4 |    0.00% |
+|                                     unclaimedGas,2 |    0.00% |
+|                              unregisterCandidate,1 |    0.00% |
+|                                             vote,2 |    0.00% |
+
```
Comment (shargon) [inline tests/Neo.SmartContract.Testing.UnitTests/Coverage/CoverageDataTests.cs:41]: @Jim8y I know you can improve this format :)

**Comment by shargon** [comment]
@Jim8y @cschuchardt88 ready to merge?

**Review by cschuchardt88 — APPROVED** [review]

**Pull request merged**